### PR TITLE
Field-coordinator and FieldOfficerId Mandatory Values

### PR DESCRIPTION
**Pull request by LeoKHoward**
# Motivation and Context
Both have become mandatory with an expected format

# What has changed
Changed to mandatory
Generates correct format when sample is made
Validators
Test

# How to test?
Check all tests pass

# Links
https://collaborate2.ons.gov.uk/confluence/pages/viewpage.action?spaceKey=CRM&title=Address+Frame+specification+2021
https://trello.com/c/oMBjUjR7/1550-field-officer-co-ord-should-be-mandatory-in-sample-validator-5

### CODE REVIEW

**Comment by LeoKHoward** [comment]
[field officer/co-ord should be mandatory in sample validator (5)](https://trello.com/c/oMBjUjR7/1550-field-officer-co-ord-should-be-mandatory-in-sample-validator-5)

File: generate_sample_file.py [hunk 1]
```diff
@@ -101,9 +101,23 @@ def get_random_post_town(self):
...
+               f'{self.get_random_letter()}{self.get_random_letter()}{first_random_number}-' \
+               f'{self.get_random_letter()}{self.get_random_letter()}'
+
+    def get_random_field_officer_id(self):
+        first_random_number = random.randint(1, 9)
+        second_random_number = random.randint(1, 9)
+        third_random_number = random.randint(1, 9)
+        return f'{self.get_random_letter()}{self.get_random_letter()}-{self.get_random_letter()}' \
+               f'{self.get_random_letter()}{self.get_random_letter()}{first_random_number}-' \
+               f'{self.get_random_letter()}{self.get_random_letter()}-{second_random_number}{third_random_number}'
```
Comment (NickGrantX) [inline generate_sample_file.py:119]: second and third number could be combined in the range 11 to 99... but no need to change it for something so minor

File: generate_sample_file.py [hunk 2]
```diff
@@ -101,9 +101,23 @@ def get_random_post_town(self):
     def get_random_post_code(self):
         first_random_number = random.randint(1, 9)
         second_random_number = random.randint(1, 9)
-        return f'{self.get_random_letter()}{self.get_random_letter()}{first_random_number} {second_random_number}' + \
+        return f'{self.get_random_letter()}{self.get_random_letter()}{first_random_number} {second_random_number}' \
                f'{self.get_random_letter()}{self.get_random_letter()}'
 
+    def get_random_field_coordinator_id(self):
```
Comment (NickGrantX) [inline generate_sample_file.py:107]: why not use this function in `get_random_field_officer_id` to save code duplication? Anyway... just minor stuff

File: validators.py [hunk 3]
```diff
@@ -138,6 +138,15 @@ def validate(postcode, **_kwargs):
     return validate
 
 
+def alphanumeric_field_values():
```
Comment (NickGrantX) [inline validators.py]: I would actually call this `alphanumeric_plus_hyphen` to be clearer what it does, although the 'field' part differentiates it I suppose

**Review by NickGrantX — APPROVED** [review]
Minor comments I'd like to see addressed, but no reason to block the PR. Great work

File: generate_sample_file.py (hunk 2 above)
Comment (LeoKHoward) [reply generate_sample_file.py:107]: In my mind, I wanted to distinguish the difference between the different formats of the two fields

**Review by AdamHawtin — APPROVED** [review]

**Pull request merged**